app/src/test/java/com/spascoding/taskycourse/feature_auth/domain/util/UserDataValidatorTest.kt

```diff
@@ -0,0 +1,68 @@
+package com.spascoding.taskycourse.feature_auth.domain.util
+
+import org.junit.Assert.assertEquals
+import org.junit.Test
+import com.spascoding.taskycourse.core.data.Result
+
+class UserDataValidatorTest {
+    @Test
+    fun `validateName returns TOO_SHORT error for names shorter than 4 characters`() {
+        val result = UserDataValidator.validateName("abc")
+        assertEquals(Result.Error(UserDataValidator.NameError.TOO_SHORT), result)
```

Comment:
Have you looked into using an assertion library? I like AssertK, makes the assertions a lot more readable:
```
assertThat(result).isEqualTo(Result.Error(UserDataValidator.NameError.TOO_SHORT))
```
Reads like a book and also comes with lots more utility functions for assertions 😄